scipy/stats/tests/test_stats.py

```diff
@@ -6010,26 +6010,35 @@ def test_describe_empty(self):
         assert_raises(ValueError, stats.describe, [])
 
 
-def test_normalitytests():
+@pytest.mark.skip_xp_backends(cpu_only=True,
+                              reasons=['Uses NumPy for pvalue'])
+@pytest.mark.usefixtures("skip_xp_backends")
+@array_api_compatible
+def test_normalitytests(xp):
     assert_raises(ValueError, stats.skewtest, 4.)
     assert_raises(ValueError, stats.kurtosistest, 4.)
     assert_raises(ValueError, stats.normaltest, 4.)
 
     # numbers verified with R: dagoTest in package fBasics
-    st_normal, st_skew, st_kurt = (3.92371918, 1.98078826, -0.01403734)
-    pv_normal, pv_skew, pv_kurt = (0.14059673, 0.04761502, 0.98880019)
+    st_normal, st_skew, st_kurt = (3.92371918, xp.asarray(1.98078826), -0.01403734)
+    pv_normal, pv_skew, pv_kurt = (0.14059673, xp.asarray(0.04761502), 0.98880019)
```

Comment:
I'm curious whether the tests pass with tighter tolerances (the default proposed in gh-20595, `4*eps**0.5`) when the values are recorded with higher precision.
```suggestion
    st_normal, st_skew, st_kurt = (3.92371918, xp.asarray(1.98078826090875881), -0.01403734)
    pv_normal, pv_skew, pv_kurt = (0.14059673, xp.asarray(0.04761502382843208), 0.98880019)
```

Let's include the R code as a comment, since the documentation of dagoTest was not really clear about how the statistic and p-value were stored.
```R
library(fBasics)
options(digits=16)
x = c(-2, -1, 0, 1, 2, 3)**2
x = rep(x, times=4)
test_result <- dagoTest(x)
test_result@test$statistic
test_result@test$p.value
```
```
      Chi2 | Omnibus       Z3  | Skewness       Z4  | Kurtosis 
 3.92371918158185551  1.98078826090875881 -0.01403734404759738 
      Omnibus  Test       Skewness Test       Kurtosis Test 
0.14059672529747502 0.04761502382843208 0.98880018772590561 
```